upup/pkg/fi/cloudup/awsup/aws_cloud.go

```diff
@@ -325,6 +336,10 @@ func NewEC2Filter(name string, values ...string) *ec2.Filter {
 
 // DeleteGroup deletes an aws autoscaling group
 func (c *awsCloudImplementation) DeleteGroup(g *cloudinstances.CloudInstanceGroup) error {
+	if c.spotinst != nil {
```

Comment:
This is going to make it hard to "graduate" spotinst from using a feature-flag - we'll probably need a field somewhere.  But we can get this in and then figure it out..  One option that might help is that AWSCloud is an interface, so it might be possible to have a spotinst implementation (even if it delegated most of the functionality to an awsCloudImplementation)

Reply:
I agree to @justinsb here - flagging spotinst to be active is one way of using it, but in the future it would be nice to just have it as a separate provider you select, that then falls back to the cloud if no special implementation was given. From what I understand you just use your own autoscaling implementation, but that also won't change that much for other clouds as you abstract that away in your own api's?

Reply:
@chrisz100 You're right about our implementation. I'm not sure I'm understanding your suggestion. How it differs from the [implementation](https://github.com/spotinst/kubernetes-kops/blob/84697892218b58cf2755339a182f87cbdd6b9333/pkg/resources/spotinst/cloud.go#L46-L61) introduced in #5195 (where we just wrapped the AWS implementation)?